filc/cmds.go

```diff
@@ -131,74 +145,260 @@ var makeDealCmd = &cli.Command{
 			return err
 		}
 
-		resp, err := fc.SendProposal(cctx.Context, proposal)
+		proto, err := fc.DealProtocolForMiner(cctx.Context, miner)
 		if err != nil {
 			return err
 		}
 
-		tpr("response state: %d", resp.Response.State)
-		switch resp.Response.State {
-		case storagemarket.StorageDealError:
-			return fmt.Errorf("error response from miner: %s", resp.Response.Message)
-		case storagemarket.StorageDealProposalRejected:
-			return fmt.Errorf("deal rejected by miner: %s", resp.Response.Message)
+		tpr("storage provider supports deal protocol %s", proto)
+
+		isPushTransfer := proto == filclient.DealProtocolv110
+		var announceAddr multiaddr.Multiaddr
+		if !isPushTransfer {
+			tpr("filc host addr: %s", nd.Host.Addrs())
+			tpr("filc host peer: %s", nd.Host.ID())
+			announce := cctx.String("announce")
+			if announce == "" {
+				return fmt.Errorf("must specify announce address to make deals over deal protocol %s", proto)
+			}
+
+			announceStr := announce + "/p2p/" + nd.Host.ID().String()
+			announceAddr, err = multiaddr.NewMultiaddr(announceStr)
+			if err != nil {
+				return fmt.Errorf("parsing announce address '%s': %w", proto, err)
+			}
+			tpr("filc announce address: %s", announceAddr.String())
+		}
+
+		dbid := uint(rand.Uint32())
```

Comment:
there is a lot of logic directly in this function that feels like its breaking abstractions of some kind, large blocks of `if !condition` smells funky

Reply:
Yeah good call that code is a bit convoluted 👍 
I refactored to split it into `makev110Deal` and `makev120Deal` and it simplifies the logic a lot. Please take a look and let me know if that approach makes sense to you. 
(I suggest viewing the diff with "Hide Whitespace" checked)
<img width="284" alt="Screen Shot 2022-03-22 at 10 39 12 AM" src="https://user-images.githubusercontent.com/169124/159451667-899ac6af-c42b-4bbd-8c5d-16ea83a28855.png">